feat: drift edit time and date action

bwees opened the pull request.

bwees committed:
feat: drift edit time and date action

bwees committed:
feat: add edit button on asset viewer bottom sheet

wuzihao051119 commented on mobile/lib/infrastructure/repositories/remote_asset.repository.dart:
```diff
@@ -186,6 +186,23 @@ class RemoteAssetRepository extends DriftDatabaseRepository {
+        );
+        batch.update(
+          _db.remoteAssetEntity,
+          RemoteAssetEntityCompanion(createdAt: Value(dateTime)),
```
```suggestion
          RemoteAssetEntityCompanion(localDateTime: Value(dateTime)),
```

bwees committed:
update localDateTime column in addition to createdAt to keep consistency

bwees committed:
fix: dont update local dateTime

Server calcs this anyway and it will be synced when the change is applied. We don't use localDateTime on mobile so there is no reason to update this value

shenlong-tanwen commented on mobile/lib/presentation/widgets/asset_viewer/bottom_sheet.widget.dart:
```diff
@@ -140,12 +141,22 @@ class _AssetDetailBottomSheet extends ConsumerWidget {
           title: _getDateTime(context, asset),
           titleStyle: context.textTheme.bodyMedium?.copyWith(fontWeight: FontWeight.w600),
+          trailing: IconButton(
+            icon: const Icon(Icons.edit, size: 18),
```
There is an additional padding around the IconButton by default. Can you add this to remove it so the edit icon is properly right aligned?
```suggestion
            constraints: const BoxConstraints(),
            padding: EdgeInsets.zero,
            style: const ButtonStyle(
              tapTargetSize: MaterialTapTargetSize.shrinkWrap, // the '2023' part
            ),
            icon: const Icon(Icons.edit, size: 18),
```

bwees committed:
fix: padding around edit icon in ListTile

Co-authored-by: shenlong <139912620+shenlong-tanwen@users.noreply.github.com>

bwees committed:
chore: format

bwees committed:
fix: hide date edit control when asset does not have a remote

bwees committed:
fix: pull timezones correctly from image
